src/utils/portfolioCalculations/getDailyPortfolio.ts

```diff
@@ -0,0 +1,43 @@
+import { CurrentPortfolioItem } from '@/types/currentPortfolio';
+import { dailyPortfolioCalculationParams } from '@/types/calculations';
+import { getCurrentPortfolio } from './getCurrentPortfolio';
+
+export type DailyPortfolio = Record<string, CurrentPortfolioItem[]>;
+
+export const getDailyPortfolio = ({
+  existingTransactions,
+  savedTransactions,
+  startDate,
+  endDate,
+}: dailyPortfolioCalculationParams): DailyPortfolio => {
+  const transactions = [...existingTransactions, ...savedTransactions];
+
+  const dateRange = Array.from(
+    {
+      length:
+        (new Date(endDate).getTime() - new Date(startDate).getTime()) /
+          (1000 * 60 * 60 * 24) +
+        1,
+    },
+    (_, i) =>
+      new Date(new Date(startDate).getTime() + i * (1000 * 60 * 60 * 24))
+        .toISOString()
+        .split('T')[0],
+  );
```

Comment:
This range calculation is not the easiest to understand. Some comments about it's purpose would be nice :)